suite-native/module-connect-device/src/components/ConnectDeviceScreenHeader.tsx

```diff
@@ -29,26 +38,68 @@ export const ConnectDeviceScreenHeader = ({
 }: ConnectDeviceScreenHeaderProps) => {
     const navigation = useNavigation<NavigationProp>();
     const dispatch = useDispatch();
+    const { translate } = useTranslate();
+    const { showAlert, hideAlert } = useAlert();
 
     const device = useSelector(selectDevice);
     const deviceModel = useSelector(selectDeviceModel);
+    const isDiscoveryActive = useSelector(selectIsDeviceDiscoveryActive);
 
-    const handleCancel = () => {
-        TrezorConnect.cancel('pin-cancelled');
-        dispatch(
-            removeButtonRequests({
-                device,
-                buttonRequestCode:
-                    deviceModel === DeviceModelInternal.T1B1
-                        ? 'PinMatrixRequestType_Current'
-                        : 'ButtonRequest_PinEntry',
-            }),
-        );
+    const handleCancel = useCallback(() => {
+        if (isDiscoveryActive) {
+            // Do not allow to cancel PIN entry while discovery is in progress
+            showAlert({
+                title: translate('moduleConnectDevice.pinCanceledDuringDiscovery.title'),
+                description: translate('moduleConnectDevice.pinCanceledDuringDiscovery.subtitle'),
+                icon: 'warningCircleLight',
+                pictogramVariant: 'red',
+                primaryButtonTitle: translate(
+                    'moduleConnectDevice.pinCanceledDuringDiscovery.button',
+                ),
+                onPressPrimaryButton: () => hideAlert(),
+            });
+        } else {
+            TrezorConnect.cancel('pin-cancelled');
 
-        if (navigation.canGoBack()) {
-            navigation.goBack();
+            dispatch(
+                removeButtonRequests({
+                    device,
+                    buttonRequestCode:
+                        deviceModel === DeviceModelInternal.T1B1
+                            ? 'PinMatrixRequestType_Current'
+                            : 'ButtonRequest_PinEntry',
+                }),
+            );
+            if (navigation.canGoBack()) {
+                navigation.goBack();
+            }
         }
-    };
+    }, [
+        device,
+        deviceModel,
+        dispatch,
+        isDiscoveryActive,
+        navigation,
+        showAlert,
+        hideAlert,
+        translate,
+    ]);
+
+    // Handle hardware back button press same as cancel button
+    useFocusEffect(
+        useCallback(() => {
+            const subscription = BackHandler.addEventListener('hardwareBackPress', () => {
+                handleCancel();
+
+                return true;
+            });
+
+            return () => {
+                subscription.remove();
+                hideAlert();
```

Comment:
this is maybe controversial? It's here to close hide the alert if PIN is entered on the device while alert is shown

Reply:
I think this is fine. Hiding alert happens in similar way in other places as well. Limitation of our alert implementation, so I'd say it's okay